src/presets/cloudflare/utils.ts

```diff
@@ -161,7 +162,7 @@ async function writeCFPagesRedirects(nitro: Nitro) {
   )) {
     const code = routeRules.redirect!.statusCode;
     contents.unshift(
-      `${key.replace("/**", "/*")}\t${routeRules.redirect!.to}\t${code}`
+      `${withBase(key.replace("/**", "/*"), nitro.options.baseURL)}\t${withBase(routeRules.redirect!.to, nitro.options.baseURL)}\t${code}`
```

Comment:
Here this is more complex as redirect might be an absolute URL so with base works best in this scenario compared to joinURL...

Reply:
I think it is same to advice: rules (including redirects) should be unware of base and we add base regardless (otherwise we cannot allow first segment of any URL to be same as dynamic base URL)

Reply:
cff5807527ee24b6759be9a64cbd49eeb5b14cb5